tests/functional/test_exclude_newer.py

```diff
@@ -0,0 +1,93 @@
+"""Tests for pip install --exclude-newer-than."""
+
+from __future__ import annotations
+
+import pytest
+
+from tests.lib import PipTestEnvironment, TestData
+
+
+class TestExcludeNewer:
+    """Test --exclude-newer-than functionality."""
+
+    def test_exclude_newer_than_invalid_date(
+        self, script: PipTestEnvironment, data: TestData
+    ) -> None:
+        """Test that --exclude-newer-than fails with invalid date format."""
+        result = script.pip(
+            "install",
+            "--no-index",
+            "-f",
+            data.packages,
+            "--exclude-newer-than=invalid-date",
+            "simple",
+            expect_error=True,
+        )
+
+        # Should fail with date parsing error
+        assert "invalid" in result.stderr.lower() or "error" in result.stderr.lower()
+
+    def test_exclude_newer_than_help_text(self, script: PipTestEnvironment) -> None:
+        """Test that --exclude-newer-than appears in help text."""
+        result = script.pip("install", "--help")
+        assert "--exclude-newer-than" in result.stdout
+        assert "datetime" in result.stdout
+
+    @pytest.mark.parametrize("command", ["install", "download", "wheel"])
+    def test_exclude_newer_than_available_in_commands(
+        self, script: PipTestEnvironment, command: str
+    ) -> None:
+        """Test that --exclude-newer-than is available in relevant commands."""
+        result = script.pip(command, "--help")
+        assert "--exclude-newer-than" in result.stdout
```

Comment:
OK, I appreciate the extensive testing, but I will take issue with these tests. I see the value, but I do not think it is worth the test execution time (remember that functional tests are slow since they need to invoke new pip subprocesses) to double check our CLI help or we've added the option to the right commands. Especially since it's not going to help if we add a new command down the line.

Reply:
Makes sense, I purposefully added too many tests and then whittled them down, I missed this one.